cocos/platform/ohos/CCDevice-ohos.cpp

```diff
@@ -0,0 +1,112 @@
+#include "platform/CCPlatformConfig.h"
+#if CC_TARGET_PLATFORM == CC_PLATFORM_OHOS
+#include "napi/helper/Js_Cocos2dxHelper.h"
+#include "platform/CCDevice.h"
+#include "base/ccTypes.h"
+#include "CCTextBitmap.h"
+#include "napi/helper/NapiHelper.h"
+
+NS_CC_BEGIN
+
+int Device::getDPI()
+{
+    return JSFunction::getFunction("DeviceUtils.getDpi").invoke<int>();
+}
+
+void Device::setAccelerometerEnabled(bool isEnabled)
+{
+    if (isEnabled)
+    {
+        Js_Cocos2dxHelper::enableAccelerometer();
+    }
+    else
+    {
+        Js_Cocos2dxHelper::disableAccelerometer();
+    }
+}
+
+void Device::setAccelerometerInterval(float interval)
+{
+    Js_Cocos2dxHelper::setAccelerometerInterval(interval);
+}
+
+class BitmapDC
+{
+public:
+
+    BitmapDC()
+    : m_nWidth(0)
+    , m_nHeight(0)
+    , m_pData(NULL)
+    {
+    }
+
+    ~BitmapDC(void)
+    {
+        if (m_pData)
+        {
+            delete [] m_pData;
+        }
+    }
+   
+    bool getBitmapWithDrawing(	const char *text, int nWidth, int nHeight, Device::TextAlign eAlignMask, const FontDefinition& textDefinition)
+    {
+        CCTextBitmap *cCtextBitmap = new CCTextBitmap();
+        CCTextBitmap::createCCTextBitmap(cCtextBitmap, text, textDefinition._fontName.data(), textDefinition._fontAlpha, textDefinition._fontFillColor.r, 
+            textDefinition._fontFillColor.g, textDefinition._fontFillColor.b ,eAlignMask, nWidth, nHeight, textDefinition._fontSize);
+        void* pixels = cCtextBitmap->getPixelAddr();
+        cocos2d::BitmapDC& bitmapDC = sharedBitmapDC();
+        bitmapDC.m_nWidth = cCtextBitmap->GetWidth();
+        bitmapDC.m_nHeight = cCtextBitmap->GetHeight();
+        long size = bitmapDC.m_nWidth * bitmapDC.m_nHeight * 4;
+        bitmapDC.m_pData = (unsigned char*)malloc(sizeof(unsigned char) * size);
```

Comment:
m_pData is released by `delete [] m_pData;` in `~BitmapDC`. Why change to `malloc` ?